### PR TITLE
ROX-16724: Operator testing on GKE

### PR DESCRIPTION
**Pull request by porridge**
## Description

* Introduce a gke-operator e2e job.
* Use a modified `kuttl` with support for conditional inclusion of files, see https://github.com/kudobuilder/kuttl/pull/483
* Split openshift-specific parts/variants of tests into separate files to make the test suite pass on GKE.
* The above broke validation of some files. Made the script skip them as a quick workaround. Proper solution tracked in a sister subtask.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Ran the tests on my workstation against an openshift and GKE cluster. Otherwise CI tests should be sufficient.

### CODE REVIEW

**Comment by porridge** [comment]
FTR, waiting for https://github.com/openshift/release/pull/38750 to be merged before testing this end-to-end.

**Comment by vikin91** [comment]
Hey @porridge , does the `gke-operator-e2e-tests` target work on master, or are you first adding it here?

**Comment by porridge** [comment]
> Hey @porridge , does the `gke-operator-e2e-tests` target work on master, or are you first adding it here?

Correct, they will only start working once this PR is complete and merged.

File: operator/README.md [hunk 1]
```diff
@@ -255,7 +255,7 @@ make kuttl
 
 These make targets will add the executable to your `$GOPATH`.
 If that is not on your `$PATH`, then you can install the Operator SDK from its [release page](https://github.com/operator-framework/operator-sdk/releases)
-and kuttl from its [release page](https://github.com/kudobuilder/kuttl/releases/tag/v0.15.0).
+and kuttl from its [release page](https://github.com/porridge/kuttl/releases).
```
Comment (ludydoo) [inline operator/README.md:258]: Why this change?

File: operator/tests/central/basic-central/10-assert.yaml [hunk 2]
```diff
@@ -1,5 +1,10 @@
 apiVersion: kuttl.dev/v1beta1
 kind: TestAssert
+commands:
+- script: |
+    set -eu # shell in CI does not grok -o pipefail
+    # On OpenShift, test that Central auth reader rolebinding exists in kube-system.
+    if kubectl get scc > /dev/null 2>&1; then kubectl get rolebinding rhacs-central-auth-reader-${NAMESPACE} -n kube-system; fi
```
Comment (ludydoo) [inline operator/tests/central/basic-central/10-assert.yaml:7]: Is this testing that the `central-auth-reader-${NS}` exists iif the `SecurityContextConstraint` CR exists ?

File: operator/tests/central/basic-central/80-assert-non-openshift.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.namespace
+            - name: ROX_TELEMETRY_ENDPOINT
+              value: https://telemetry.endpoint
+            - name: ROX_TELEMETRY_STORAGE_KEY_V1
+              value: my-api-key
+            - name: ROX_OFFLINE_MODE
+              value: "false"
```
Comment (ludydoo) [inline operator/tests/central/basic-central/80-assert-non-openshift.yaml:43]: Is this needed since `ROX_OFFLINE_MODE` seems to be false by default ?

File: operator/README.md (hunk 1 above)
Comment (porridge) [reply operator/README.md:258]: See PR description:

> Use a modified kuttl with support for conditional inclusion of files, see https://github.com/kudobuilder/kuttl/pull/483

That change was merged upstream by now, but it might still be a while until we see a new release.

File: operator/tests/central/basic-central/10-assert.yaml (hunk 2 above)
Comment (porridge) [reply operator/tests/central/basic-central/10-assert.yaml:7]: Yes, exit code 0 from `kubectl get scc` is apparently the idiomatic way of checking whether this is an OpenShift cluster.

File: operator/tests/central/basic-central/80-assert-non-openshift.yaml (hunk 3 above)
Comment (porridge) [reply operator/tests/central/basic-central/80-assert-non-openshift.yaml:43]: Not sure if that's what you're asking, but yes, it's needed, otherwise the test would fail because of very rudimental support for arrays in kuttl.

**Review by ludydoo — APPROVED** [review]
To me it looks all good, though I have a limited context on e2e tests

**Comment by porridge** [comment]
Thanks @ludydoo ! Let me know if https://github.com/stackrox/stackrox/blob/master/operator/tests/TROUBLESHOOTING_E2E_TESTS.md is helpful, or if you'd like a live introduction to this topic.

Rebasing to make the style check pass...
